TypeError: is not assignable to type TransportStream writable missing

```
error TS2322: Type '(DailyRotateFile | ConsoleTransportInstance)[]' is not assignable to type 'TransportStream | TransportStream[]'.
  Type '(DailyRotateFile | ConsoleTransportInstance)[]' is not assignable to type 'TransportStream[]'.
    Type 'DailyRotateFile | ConsoleTransportInstance' is not assignable to type 'TransportStream'.
      Type 'DailyRotateFile' is not assignable to type 'TransportStream'.
        Property 'writable' is missing in type 'DailyRotateFile'.
```

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,57 +1,63 @@
-import * as Transport from 'winston-transport';
-declare namespace DailyRotateFile {
-    interface DailyRotateFileTransportOptions extends Transport.TransportStreamOptions {
-      json?: boolean;
-      eol?: string;
-  
-      /**
-       * A string representing the moment.js date format to be used for rotating. The meta characters used in this string will dictate the frequency of the file rotation. For example, if your datePattern is simply 'HH' you will end up with 24 log files that are picked up and appended to every day. (default 'YYYY-MM-DD')
-       */
-      datePattern?: string;
-  
-      /**
-       * A boolean to define whether or not to gzip archived log files. (default 'false')
-       */
-      zippedArchive?: boolean;
-  
-      /**
-       * Filename to be used to log to. This filename can include the %DATE% placeholder which will include the formatted datePattern at that point in the filename. (default: 'winston.log.%DATE%)
-       */
-      filename?: string;
-  
-      /**
-       * The directory name to save log files to. (default: '.')
-       */
-      dirname?: string;
-  
-      /**
-       * Write directly to a custom stream and bypass the rotation capabilities. (default: null)
-       */
-      stream?: NodeJS.WritableStream;
-  
-      /**
-       * Maximum size of the file after which it will rotate. This can be a number of bytes, or units of kb, mb, and gb. If using the units, add 'k', 'm', or 'g' as the suffix. The units need to directly follow the number. (default: null)
-       */
-      maxSize?: string | number;
-  
-      /**
-       * Maximum number of logs to keep. If not set, no logs will be removed. This can be a number of files or number of days. If using days, add 'd' as the suffix. (default: null)
-       */
-      maxFiles?: string | number;
-  
-      /**
-       * An object resembling https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options indicating additional options that should be passed to the file stream. (default: `{ flags: 'a' }`)
-       */
-      options?: string | object;
-    }
+declare module "winston-daily-rotate-file" {
+  import * as Transport from "winston-transport";
+
+  namespace DailyRotateFile {
+      interface DailyRotateFileTransportOptions extends Transport.TransportStreamOptions {
+          json?: boolean;
+          eol?: string;
+
+          /**
+           * A string representing the moment.js date format to be used for rotating. The meta characters used in this string will dictate the frequency of the file rotation. For example, if your datePattern is simply 'HH' you will end up with 24 log files that are picked up and appended to every day. (default 'YYYY-MM-DD')
+           */
+          datePattern?: string;
+
+          /**
+           * A boolean to define whether or not to gzip archived log files. (default 'false')
+           */
+          zippedArchive?: boolean;
+
+          /**
+           * Filename to be used to log to. This filename can include the %DATE% placeholder which will include the formatted datePattern at that point in the filename. (default: 'winston.log.%DATE%)
+           */
+          filename?: string;
+
+          /**
+           * The directory name to save log files to. (default: '.')
+           */
+          dirname?: string;
+
+          /**
+           * Write directly to a custom stream and bypass the rotation capabilities. (default: null)
+           */
+          stream?: NodeJS.WritableStream;
+
+          /**
+           * Maximum size of the file after which it will rotate. This can be a number of bytes, or units of kb, mb, and gb. If using the units, add 'k', 'm', or 'g' as the suffix. The units need to directly follow the number. (default: null)
+           */
+          maxSize?: string | number;
+
+          /**
+           * Maximum number of logs to keep. If not set, no logs will be removed. This can be a number of files or number of days. If using days, add 'd' as the suffix. (default: null)
+           */
+          maxFiles?: string | number;
+
+          /**
+           * An object resembling https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options indicating additional options that should be passed to the file stream. (default: `{ flags: 'a' }`)
+           */
+          options?: string | object;
+      }
   }
-  
-  declare class DailyRotateFile extends Transport {
-    filename: string;
-    dirname: string;
-    logStream: NodeJS.WritableStream;
-    options: DailyRotateFile.DailyRotateFileTransportOptions;
-    constructor (options?: DailyRotateFile.DailyRotateFileTransportOptions);
+
+  interface DailyRotateFileTransportInstance extends Transport {
+      filename: string;
+      dirname: string;
+      logStream: NodeJS.WritableStream;
+      options: DailyRotateFile.DailyRotateFileTransportOptions;
+
+      new (options?: DailyRotateFile.DailyRotateFileTransportOptions): DailyRotateFileTransportInstance;
   }
-  
+
+  const DailyRotateFile: DailyRotateFileTransportInstance;
+
   export = DailyRotateFile;
+}

